src/vs/workbench/browser/parts/editor/editorDropTarget.ts

```diff
@@ -292,6 +296,24 @@ class DropOverlay extends Themable {
 			}
 		}
 
+		// Check for tree items
+		else if (this.treeItemsTransfer.hasData(DraggedExtensionTreeItemsIdentifier.prototype)) {
+			const data = this.treeItemsTransfer.getData(DraggedExtensionTreeItemsIdentifier.prototype);
+			if (Array.isArray(data)) {
+				const treeData = Promise.all(
+					data.map(id => this.treeViewsDragAndDropService.removeDragOperationTransfer(id.identifier)));
+				treeData.then(dataTransferItems => {
+					const dropHandler = this.instantiationService.createInstance(ResourcesDropHandler, { allowWorkspaceOpen: true /* open workspace instead of file if dropped */ });
+					dataTransferItems.forEach(dataTransferItem => {
```

Comment:
I find this a bit weird tbh: the original idea of `ResourcesDropHandler` was to only deal with `DropEvent`, basically a catch all handler if we did not find a local selection transfer. Can we not extract the editors from the treeItemsTransfer right here in this location?

Besides, would it make sense to not loop over each item and handle it but first resolve all editors and then open them all at once?

Reply:
Moved into dnd.ts.